app/controllers/participants/validations_controller.rb

```diff
@@ -0,0 +1,224 @@
+      end
+    end
+
+    def find_your_trn; end
+
+    def get_a_trn; end
+
+    def have_you_changed_your_name
+      choice = @participant_validation_form.have_you_changed_your_name_choice
+      if choice == "yes"
+        store_form_and_redirect_to_step :confirm_updated_record
+      else
+        store_form_and_redirect_to_step :tell_us_your_details
+      end
+    end
+
+    def confirm_updated_record
+      choice = @participant_validation_form.updated_record_choice
+      if choice == "yes"
+        store_form_and_redirect_to_step :tell_us_your_details
+      elsif choice == "no"
+        store_form_and_redirect_to_step :name_not_updated
+      else
+        store_form_and_redirect_to_step :check_with_tra
+      end
+    end
+
+    def name_not_updated
+      choice = @participant_validation_form.name_not_updated_choice
+      if choice == "register_previous_name"
+        store_form_and_redirect_to_step :tell_us_your_details
+      else
+        store_form_and_redirect_to_step :change_your_details_with_tra
+      end
+    end
+
+    def change_your_details_with_tra; end
+
+    def check_with_tra; end
+
+    def tell_us_your_details
+      store_form_and_redirect_to_step :confirm_details
+    end
+
+    def confirm_details
+      validate_participant_details_and_redirect
+    end
+
+    def cannot_find_details
+      store_validation_data!
+      reset_form_data
+      store_form_and_redirect_to_step :complete
+    end
+
+    def complete
+      @school = participant_profile.school
+      if participant_profile.ecf_participant_eligibility&.eligible_status?
+        # TRN has been validated, qts, no flags, not done before, no previous induction
+        render_completed_page
+      else
+        # all other cases
+        render_manual_check_page
+      end
+    end
+
+    def reset
+      reset_form_data
+      redirect_to participants_validation_start_path
+    end
+
+  private
+
+    def validate_request_or_render
+      render unless request.put? && step_valid?
+    end
+
+    def render_completed_page
+      if participant_profile.school_cohort.full_induction_programme?
+        @partnership = @school.partnerships.active.find_by(cohort: participant_profile.school_cohort.cohort)
+        render "complete_fip"
+      else
+        render "complete_cip"
+      end
+    end
+
+    def render_manual_check_page
+      if participant_profile.school_cohort.full_induction_programme?
+        @partnership = @school.partnerships.active.find_by(cohort: participant_profile.school_cohort.cohort)
+        render "manual_details_check_fip"
+      else
+        render "manual_details_check_cip"
+      end
+    end
+
+    def check_not_already_completed
+      store_form_and_redirect_to_step :complete if flow_complete?
+    end
+
+    def flow_complete?
+      participant_profile.ecf_participant_validation_data.present? || participant_profile.ecf_participant_eligibility.present?
+    end
+
+    def validate_participant_details_and_redirect
```

Comment:
I'm not sure what the timeout is set to on the API client at the moment, might be worth checking at some point